src/symbol/placement.js

```diff
@@ -256,6 +256,9 @@ export class Placement {
                     offscreen = offscreen && placedIconBoxes.offscreen;
                 }
 
+                const iconWithoutText = textOptional || (symbolInstance.numGlyphVertices === 0 && symbolInstance.numVerticalGlyphVertices === 0);
```

Comment:
These variable names tripped me up quite a bit. Would `textInstanceOptional` and `iconInstanceOptional` make more sense?